### PR TITLE
Improved Money Printer

### PR DESCRIPTION
**Pull request by 00875**
Now it can be base for others money printers
Added config Config.printerreward
Fixed stopping sounds on sounds cleanup
FUCKING SPACES

### CODE REVIEW

File: entities/entities/money_printer/init.lua [hunk 1]
```diff
@@ -77,52 +87,39 @@ function ENT:Fireball()
...
-    timer.Simple(3, function()
-        if not IsValid(ent) then return end
-        ent:CreateMoneybag()
-    end)
-end
-
 function ENT:CreateMoneybag()
     if not IsValid(self) or self:IsOnFire() then return end
-
+
```
Comment (FPtje) [inline entities/entities/money_printer/init.lua]: Trailing whitespace

File: entities/entities/money_printer/shared.lua [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
 ENT.Base = "base_gmodentity"
 ENT.PrintName = "Money Printer"
-ENT.Author = "Render Case and philxyz"
+ENT.Author = "DarkRP Developers"
 ENT.Spawnable = false
+ENT.sparking = false
+ENT.IsMoneyPrinter = true
+
+--Vars
+ENT.MoneyCount = GAMEMODE.Config.mprintamount
```
Comment (FPtje) [inline entities/entities/money_printer/shared.lua]: Like @Bo98 's situation with the common lab, I think these variables will be set before the DarkRPMod's config files are run.

File: entities/entities/money_printer/init.lua (hunk 1 above)
Comment (FPtje) [inline entities/entities/money_printer/init.lua]: I recommend installing sublime text, glualint and setting this option to true:

```json
"trim_trailing_white_space_on_save": true
```

File: entities/entities/money_printer/init.lua [hunk 3]
```diff
@@ -134,7 +143,7 @@ function ENT:Think()
 end
 
 function ENT:OnRemove()
-    if self.sound then
+    if IsValid(self.sound) then
```
Comment (Bo98) [inline entities/entities/money_printer/init.lua]: I don't think IsValid works on CSoundPatch (object returned by CreateSound).

File: entities/entities/money_printer/init.lua [hunk 4]
```diff
@@ -112,17 +122,16 @@ function ENT:CreateMoneybag()
...
 
 function ENT:Think()
-
     if self:WaterLevel() > 0 then
         self:Destruct()
         self:Remove()
         return
     end
-
+    if not IsValid(self.sound) then self:StartSound() end
```
Comment (Bo98) [inline entities/entities/money_printer/init.lua]: Same situation here in case you miss it.

File: entities/entities/money_printer/init.lua (hunk 3 above)
Comment (00875) [inline entities/entities/money_printer/init.lua]: It works. I using this on my server

**Comment by NamefuI** [comment]
Jesus christ, can you at least try to act like a normal person on your commits? They look like they're written by a 12 year old who learned to code lua in a tutorial out of a cereal box.
Sorry, but it's what I feel.

File: entities/entities/money_printer/init.lua [hunk 5]
```diff
@@ -1,27 +1,50 @@
...
+    end)
+end
+
+function ENT:StartSound()
+    self.sound = CreateSound(self, Sound("ambient/levels/labs/equipment_printer_loop1.wav"))
+    self.sound:SetSoundLevel(52)
+    self.sound:PlayEx(1, 100)
+end
+
+function ENT:initVars()
```
Comment (FPtje) [inline entities/entities/money_printer/init.lua]: Pretty much duplicate code with shared.lua:10 onwards. Why isn't this function shared?

**Comment by FPtje** [comment]
They're still defined twice, for fuck's sake!

**Comment by 00875** [comment]
"--Define for avoid errors". I made this to avoid fucking error reports if someone made money printer too bad

**Comment by FPtje** [comment]
That's a shit justification. All I had to do to fix your errors was move the initVars up in the Initialise hook. The labs don't have this problem, so why should yours?

**Comment by 00875** [comment]
I don't know. You know better.

**Pull request merged**